core/Pos/Binary/Merkle.hs

```diff
@@ -4,16 +4,16 @@ module Pos.Binary.Merkle () where
 
 import           Universum
 
-import           Pos.Binary.Class   (Bi (..), Raw, label, putField)
+import           Pos.Binary.Class   (Bi (..), Raw)
 import           Pos.Crypto.Hashing (Hash)
 import           Pos.Merkle         (MerkleRoot (..), MerkleTree (..), mkMerkleTree)
 
 -- This instance is both faster and more space-efficient (as confirmed by a
 -- benchmark). Hashing turns out to be faster than decoding extra data.
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleTree a) where
-    sizeNPut = putField toList
-    get = label "MerkleTree" $ mkMerkleTree <$> get
+  encode = encode . toList
+  decode = mkMerkleTree <$> decode
 
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleRoot a) where
-    sizeNPut = putField getMerkleRoot
-    get = label "MerkleRoot" $ MerkleRoot <$> get
+  encode = encode . getMerkleRoot
+  decode = MerkleRoot <$> decode
```

Comment:
As a general etiquette I always tend to honour whichever code style is in place when contributing to a new project. This time I was sloppier than usual due to the huge amount of refactoring I had to make (in other terms that went to be back of my priority list, hehe). Something I would repeat here is that it would be awesome if somebody could come up with an Emacs module (or an equivalent stuff for people using other editors) to enforce such style guide automatically upon formatting. It's quite time consuming trying to keep in my head all the different new style guides whilst fixing bugs / adding features 😉